### PR TITLE
Improve bitmap::_calculateArea performance

### PR DESCRIPTION
**Pull request by lhecker**
`bitmap::_calculateArea` performance can be improved by leveraging the
optimized `find_first`/`find_next` methods instead of iterating through
the bitmap manually.

### CODE REVIEW

File: src/inc/til/bitmap.h [hunk 1]
```diff
@@ -119,7 +113,8 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
                 else
                 {
                     // If we reached the end, set the pos because the run is empty.
-                    _pos = _nextPos;
+                    _pos = _end;
+                    _nextPos = _end;
```
Comment (lhecker) [inline src/inc/til/bitmap.h:117]: `find_next` returns `2^63-1` if it was unable to find the next-set-bit. Due to that `_nextPos > _end` is possible.

**Comment by zadjii-msft** [comment]
@msftbot make sure @miniksa signs off on this

File: src/inc/til/bitmap.h (hunk 1 above)
Comment (zadjii-msft) [reply src/inc/til/bitmap.h:117]: This seems like a valuable comment to leave in the code IMO

**Comment by lhecker** [comment]
@DHowett `dynamic_bitset::operator~` creates a deep copy of itself (including the backing buffer). I'm not sure if that's really worth it...

File: src/inc/til/bitmap.h [hunk 2]
```diff
@@ -83,14 +83,8 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
...
             {
-                // Backup the position as the next one.
-                _nextPos = _pos;
-
-                // Seek forward until we find an on bit.
-                while (_nextPos < _end && !_values[_nextPos])
-                {
-                    ++_nextPos;
-                }
+                const auto nextPos = _pos == 0 ? _values.find_first() : _values.find_next(_pos - 1);
```
Comment (miniksa) [inline src/inc/til/bitmap.h:86]: I would prefer a comment left above this explaining what it's doing in plain language. It requires a bit of mental effort to decode the ternary and remember/lookup how `find_first` and `find_next` operate. With algorithms such as these, I like to leave too much explanation of why/how they work so a reader at a later date can understand what to care about and what not to care about without having to cross-reference anything.

File: src/inc/til/bitmap.h (hunk 1 above)
Comment (miniksa) [reply src/inc/til/bitmap.h:117]: Yeah, leave that comment here and it's good for me.

**Comment by DHowett** [comment]
I guess I was thinking of the cheap case where the deep copy was only a couple uint64s. Given that the number of bits in our standard 120x30 makes it take up 57 uint64s, it’s almost certainly more expensive than iterating over the small number of bits to find a 0

**Comment by lhecker** [comment]
@miniksa I added some comments. Are they alright? 🙂

**Comment by DHowett** [comment]
Marked for automatic merge after CI

**Pull request merged**